api/controllers/UserController.js

```diff
@@ -34,33 +31,26 @@ module.exports = {
       req.session.usr = user.id;
       return res.ok(user.id);
     } catch (err) {
-      return res.badRequest(err);
+      return res.badRequest({
+        message: 'Unable to register user',
+      });
     }
   },
-  login: function (req, res) {
-    const { username } = req.body;
-    if (username) {
-      userAPI
-        .findUserByUsername(username)
-        .then((user) => {
-          return passwordAPI
-            .checkPass(req.body.password, user.encryptedPassword)
-            .then(() => {
-              req.session.loggedIn = true;
-              req.session.usr = user.id;
-              return res.ok();
-            })
-            .catch((reason) => {
-              return res.badRequest(reason);
-            });
-        })
-        .catch(() => {
-          return res.badRequest({
-            message: 'Could not find that user with that username. Try signing up!',
-          });
-        });
-    } else {
-      return res.badRequest({ message: 'A username must be provided' });
+  login: async function (req, res) {
```

Comment:
I also took the opportunity to rewrite the sails methods while I was in here to consolidate them and get them updated to async await.